### PR TITLE
Problem (Fix #2145): incorrect confirmed_transcript_hash calculation in mls

### LINKED ISSUE #2145: Problem: incorrect confirmed_transcript_hash calculation in mls
https://github.com/crypto-com/chain/blob/master/chain-tx-enclave-next/mls/src/group.rs#L395

`get_init_confirmed_transcript_hash` is only valid for the initial group state (0), as it hardcodes "" as `interim_transcript_hash`... for the rest, it should be calculated as:
```
interim_transcript_hash_[n+1] =
    Hash(confirmed_transcript_hash_[n] ||
        MLSPlaintextCommitAuthData_[n]);
```

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- update implementation according to spec

### CODE REVIEW

File: chain-tx-enclave-next/mls/src/group.rs [hunk 1]
```diff
@@ -179,24 +181,28 @@ impl<CS: CipherSuite + Ord> GroupAux<CS> {
...
             sender,
             commit.clone(),
         )
         .get_encoding();
-        let to_hash = [interim_transcript_hash, content_to_commit].concat();
+        let to_hash = [
+            self.interim_transcript_hash
+                .as_ref()
+                .map(|h| h.as_ref())
+                .unwrap_or_default(),
```
Comment (tomtau) [inline chain-tx-enclave-next/mls/src/group.rs:197]: maybe worth expanding on what is going on here / why it matches the spec -- the type is (`&Vec<u8>.as_ref()`) `&[u8]` and its default == `b""` ? is that correct?

Comment (yihuang) [reply chain-tx-enclave-next/mls/src/group.rs]: yes, changed to more explicit `unwrap_or(b"")`, and copied the spec as comment.

**Comment by yihuang** [comment]
> Build failed (retrying...):
> 
> * [drone-hw/push](http://13.82.172.109/crypto-com/chain/1222)

#2093 again

**Comment by tomtau** [comment]
bors retry

**Pull request merged**